### ISSUE TITLE
Return type inference

### ISSUE THREAD

**Issue by letrec** [microsoft/pyright#264]
**Describe the bug**
Return type of ’i‘ and ’s‘ is inferred as ‘Any‘ in some cases.

**To Reproduce**
See the snippet below.

**Expected behavior**
Return type of ’i‘ and ’s‘ is inferred as ‘int‘ and ‘str‘ respectively.

**Screenshots or Code**

```
async def test():
    import asyncio

    async def get_int() -> int:
        return 42

    async def get_str() -> str:
        return "Hi!"

    i, s = await asyncio.gather(get_int(), get_str())
```

**VS Code extension or command-line**
VS Code extension version is 1.0.65.

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the bug. I'll look into it.

**Comment by erictraut** [comment]
This was due to a recent regression in the type checker that was not caught by my unit tests. I've fixed the issue and added your example to the test cases to prevent similar regressions in the future. The fix will be in the next version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.66, which I just published.
